File: patternsjava/mvc/MVCPatternDemo.java
package patternsjava.mvc;

public class MVCPatternDemo {
    public static void main(String[] args) {

        // fetch student record based on his roll no from the database
        Student model = retriveStudentFromDatabase();

        // Create a view : to write student details on console
        StudentView view = new StudentView();

        StudentController controller = new StudentController(model, view);

        controller.updateView();

        // update model data
        controller.setStudentName("John");

        controller.updateView();
    }

    private static Student retriveStudentFromDatabase() {
        Student student = new Student();
        student.setName("Robert");
        student.setRollNo("10");
        return student;
    }
}

File: patternsphp/mvc/MVCPatternDemo.php
<?php

declare(strict_types=1);

require_once __DIR__ . '/Student.php';
require_once __DIR__ . '/StudentView.php';
require_once __DIR__ . '/StudentController.php';

class MVCPatternDemo
{
    public static function retrieveStudentFromDatabase(): Student
    {
        $student = new Student();
        $student->setName("Robert");
        $student->setRollNo("10");
        return $student;
    }
}

//fetch student record based on his roll no from the database
$model = MVCPatternDemo::retrieveStudentFromDatabase();

//Create a view : to write student details on console
$view = new StudentView();

//Create controller
$controller = new StudentController($model, $view);

$controller->updateView();

//Update model data
$controller->setStudentName("John");

$controller->updateView();
